test/custom-field-basic-test.html

```diff
@@ -122,6 +130,71 @@
           expect(customField.hasAttribute('focused')).to.be.false;
         });
       });
+
+      describe('errorMessage', function() {
+        it('setting errorMessage updates has-error-message attribute', function() {
+          customField.errorMessage = 'foo';
+          expect(customField.hasAttribute('has-error-message')).to.be.true;
+        });
+
+        it('setting errorMessage to empty string does not update has-error-message attribute', function() {
+          customField.errorMessage = '';
+          expect(customField.hasAttribute('has-error-message')).to.be.false;
+        });
+
+        it('setting errorMessage to null does not update has-error-message attribute', function() {
+          customField.errorMessage = null;
+          expect(customField.hasAttribute('has-error-message')).to.be.false;
+        });
+      });
+
+      describe('helperText', function() {
+        it('setting helper updates has-helper attribute', function() {
+          customField.helperText = 'foo';
+          expect(customField.hasAttribute('has-helper')).to.be.true;
+        });
+
+        it('setting helper to empty string does not update has-helper attribute', function() {
+          customField.helperText = '';
+          expect(customField.hasAttribute('has-helper')).to.be.false;
+        });
+
+        it('setting helper to null does not update has-helper attribute', function() {
+          customField.helperText = null;
+          expect(customField.hasAttribute('has-helper')).to.be.false;
+        });
+
+        it('setting number helper updates has-helper attribute', function() {
+          customField.helperText = 0;
+          expect(customField.hasAttribute('has-helper')).to.be.true;
+        });
+
+        it('field with slotted helper updates has-helper attribute', function() {
+          const customFieldWithSlottedHelper = fixture('default-with-slotted-helper');
+          customFieldWithSlottedHelper.__helperObserver.flush();
+          expect(customFieldWithSlottedHelper.hasAttribute('has-helper')).to.be.true;
+        });
+
+        it('setting helper with slot updates has-helper attribute', function() {
+          const helper = document.createElement('div');
+          helper.setAttribute('slot', 'helper');
+          helper.textContent = 'foo';
+          customField.appendChild(helper);
+          customField.__helperObserver.flush();
+
+          expect(customField.hasAttribute('has-helper')).to.be.true;
+        });
+
+        it('removing slotted helper removes has-helper attribute', function() {
+          const radioButtonGroupWithSlottedHelper = fixture('default-with-slotted-helper');
+
+          const helper = radioButtonGroupWithSlottedHelper.querySelector('[slot="helper"]');
+          radioButtonGroupWithSlottedHelper.removeChild(helper);
+          radioButtonGroupWithSlottedHelper.__helperObserver.flush();
+
+          expect(radioButtonGroupWithSlottedHelper.hasAttribute('has-helper')).to.be.false;
```

Comment:
We should probably also test `aria-describedby` construction with `helperText` and `errorMessage`.

Reply:
Done.